src/hooks/useTxHistory.ts

```diff
@@ -11,24 +13,30 @@ const useTxHistory = (
   error?: string
   loading: boolean
 } => {
+  const router = useRouter()
+  const hasFilter = hasTxFilterQuery(router.query)
```

Comment:
I would replace this with `filter = getTxFilter(router.query)`. Then you can just pass that to the useAsync instead of calling it every time the query changes.

Reply:
The `filter` doesn't contain the `TxFilterFormType`. We need to check the `query` every time it changes in order to determine the `TxFilterFormType` and, in turn, which endpoint to fetch from.